planner/core/task.go

```diff
@@ -1169,7 +1169,11 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 		// We should not push agg down across double read, since the data of second read is ordered by handle instead of index.
 		// The `extraHandleCol` is added if the double read needs to keep order. So we just use it to decided
 		// whether the following plan is double read with order reserved.
-		if cop.extraHandleCol == nil {
+		if cop.extraHandleCol != nil || len(cop.rootTaskConds) > 0 {
```

Comment:
Only `Aggregation` is affected? How about `TopN` and `Limit`?

Reply:
Yes, you are right.